src/test/lifecycle/common.js

```diff
@@ -512,3 +517,20 @@ exports.test_and_prefix_size = async function(Bucket, Key, s3) {
     assert(actualFilter.ObjectSizeGreaterThan === expectedFilter.ObjectSizeGreaterThan, 'and prefix size filter - ObjectSizeGreaterThan');
     assert(actualFilter.ObjectSizeLessThan === expectedFilter.ObjectSizeLessThan, 'and prefix size filter - ObjectSizeLessThan');
 };
+
+exports.test_expiration_date_in_past = async function(Bucket, Key, s3) {
+    const putLifecycleParams = date_lifecycle_configuration(Bucket, Key);
+
+    // decrease expiration date by 2 days to set it in the past
+    const date = new Date(Date.now());
+    date.setDate(date.getDate() - 2);
+    const past_date = new Date(date.setUTCHours(0, 0, 0, 0));
+    putLifecycleParams.LifecycleConfiguration.Rules[0].Expiration.Date = past_date;
```

Comment:
Maybe clone the object `putLifecycleParams` and then change it?
Because someone might write another test after and the object would be changed.

Reply:
For every test we are initializing it, so I think we are good here. Thanks

Reply:
Thanks, Where?

Reply:
you can check the other test functions in the same file `common.js` , for every function we are initializing `putLifecycleParams` first and then we are calling the put_get_lifecycle_configuration(putLifecycleParams). Pls correct me if I am wrong

Reply:
You're right, thank you.